app/ui/node-settings/browser.js

```diff
@@ -0,0 +1,310 @@
+(function () {
+  'use strict'
+  const handlebars = require('handlebars')
+  const uuid4      = require('uuid4')
+  const i18n       = require('../../lib/i18n.js')
+
+  const logger       = require('../../lib/logger.js');
+  const clientConfig = require('../../lib/config.js');
+  const syncFactory  = require('@gyselroth/balloon-node-sync');
+
+  const electron = require('electron');
+  const ipcRenderer = electron.ipcRenderer;
+
+  handlebars.registerHelper('i18n', (key) => {
+    var translation = i18n.__(key);
+
+    return new handlebars.SafeString(translation);
+  })
+
+  var sync,
+      promise = new Promise((resolve) => {
+        if (!clientConfig.get('authMethod')) {
+          logger.info('AUTH: no authentication method set yet');
+          return resolve();
+        }
+
+        clientConfig.retrieveSecret(clientConfig.getSecretType()).then((secret) => {
+          clientConfig.setSecret(secret);
+          resolve();
+        }).catch((error) => {
+          logger.error('AUTH: failed retrieve secret from keystore', {error});
+          resolve();
+        });
+      });
+
```

Comment:
You can not receive the secret like this.
the main process (your controller) needs to send the secret to the thread.
Otherwise you would end up in failures or the user gets prompted to allow access to the secret store.

controller:
```
ipcMain.once('node-settings-window-loaded', function(){
       nodeSettingsWindow.webContents.send('secret', clientConfig.getSecretType(), clientConfig.getSecret());
});
```
browser.js
```
  ipcRenderer.send('node-settings-window-loaded');
  ipcRenderer.once('secret', function(event, type, secret) {
    var config = clientConfig.getAll(true);
    config[type] = secret;
    sync = syncFactory(config, standardLogger);

    //do stuff
});
```